src/app/components/calendar/calendarForm.tsx

```diff
@@ -25,34 +26,27 @@ const CalendarForm = () => {
     color: "blackAndWhite",
   };
 
-  const [calendarSetup, setCalendar] = useState<CalendarValuesProps | null>(null);
-
-  const [calendarDates, setCalendarDates] = useState({
+  const calendarDatesDefault = {
     weekNumber: null,
     weekDayNumbers: [],
     month: new Date().getMonth(),
     year: new Date().getFullYear(),
-  });
+  }
+
+  const [calendarSetup, setCalendar] = useLocalStorage('calendarData', calendarDefaultValues)
```

Comment:
It is customary for the state variable and the state setter to have names X and setX. Eg:

```suggestion
  const [calendarSettings, setCalendarSettings] = useLocalStorage('calendarData', calendarDefaultValues)
```